### ISSUE TITLE
Roxy should support pluggable components (ie other github projects that can be pulled in)

### ISSUE THREAD

**Issue by stevenbrockman** [marklogic/roxy#565]
Idea here being, some XQuery code and libraries would get automatically deployed under: /src

Then, a XML section for ROXY-ROUTES specific to the plugin could get pulled into /app/config/config.xqy so that the new pages/controllers can be invoked.  Example:

File: **config.mycomponent.xml** that may hold something like this:
`<request uri="^/spawn/(.*)"/>`

**Comment by paxtonhare** [comment]
It sounds like you want this: https://github.com/joemfb/mlpm

-Paxton

**Comment by stevenbrockman** [comment]
Not quite.  This is a node js utility.  We were looking to pull in XQuery specific projects.  Like [Telemetry] (https://github.com/bradmann/marklogic-telemetry)  - Granted that project may need to be modified a little as well to fit into some solution.

**Comment by joemfb** [comment]
While the client is a node.js CLI tool, but the system is designed to support MarkLogic server-side modules of any kind. Here's some more info: http://registry.demo.marklogic.com/ (which I should probably incorporate into the client repo README).
